scripts/validate_merkle.ts

```diff
@@ -0,0 +1,158 @@
+const hre = require('hardhat')
+const fs = require('fs')
+const csv = require('csv-parse')
+
+const ethereumMulticall = require('ethereum-multicall')
+
+const REQUIRE_PARSE = true
+
+const PROJECT_ID = 'AD_2fSv9K1GHoUG'
+
+async function loadCSV(
+    filePath: string
+): Promise<{Recipient: string; 'Token Allocated': string}[]> {
+    const results: {Recipient: string; 'Token Allocated': string}[] = []
+
+    return new Promise((resolve, reject) => {
+        fs.createReadStream(filePath)
+            .pipe(
+                csv.parse({
+                    delimiter: ',',
+                    columns: true,
+                    ltrim: true
+                })
+            )
+            .on('data', (data: any) => results.push(data))
+            .on('end', () => {
+                resolve(results)
+            })
+            .on('error', reject)
+    })
+}
+
+async function requestProofs(
+    tokenIds: string[],
+    projectId: string
+): Promise<ClaimData[]> {
+    const myHeaders = new Headers()
+    myHeaders.append('Content-Type', 'application/json')
+
+    const raw = JSON.stringify({
+        recipients: tokenIds,
+        projectId
+    })
+
+    const requestOptions = {
+        method: 'POST',
+        headers: myHeaders,
+        body: raw
+    }
+
+    return fetch(
+        'https://moca-claim.tokentable.xyz/api/airdrop-open/batch-query',
+        requestOptions
+    )
+        .then((response) => response.json())
+        .then((res) => res.data.claims)
+}
+
+interface ClaimData {
+    recipient: string
```

Comment:
maybe out of topic - but this number from token team - token ID or name?

Reply:
its by token id, in all the script we should also just using token id not token name